libtenzir/include/tenzir/actors.hpp

```diff
@@ -297,6 +297,8 @@ using filesystem_actor = typed_actor_fwd<
   auto(atom::write, std::filesystem::path, chunk_ptr)->caf::result<atom::ok>,
   // Reads a chunk of data from a given path and returns the chunk.
   auto(atom::read, std::filesystem::path)->caf::result<chunk_ptr>,
+  // Reads all files from a given directory and returns their content as a 'blob'.
+  auto(atom::read, atom::recursive, std::filesystem::path)->caf::result<record>,
```

Comment:
This looks like a mis-use of the `record` type to me. Why not return a `map` from string to `chunk_ptr`?

Reply:
A record preserves the directory structure, with a map we'd have to reconstruct it from the keys in a post-processing step (which might not even be unambigously possible, if there are filenames containing slashes)

We could make a nested map, but I don't see the benefit of introducing a second recursive data structure when we already have one that can do the job.

Reply:
Ah, it was not clear to me from the comment that this keeps the directory structure by returning nested record fields for directories. Then let's just keep this as-is for now, although I do think it'd be better to change this eventually.